### ISSUE TITLE
Xiaomi Gateway problem when upgrade Domoticz version.

### ISSUE THREAD

**Issue by vwtune** [domoticz/domoticz#3649]
I use 2 Xiaomi gateways (and quite a few devices linked to it) in Domoticz Version: 4.9700.
Now I am forced by a 'problem' to update Domoticz to a newer (latest) version. That 'problem' has therefore been resolved, but I do get another one, which I find even worse than what I first had before I had to update. After updating I immediately get the message in the log from Domoticz: Xiaomi Gateway: receive data from unregistered gateway!
I can control both gateways (doorbell or lighting) but the connected devices no longer work on 1 gateway. I have already tried to deactivate them 1 by 1, restart domoticz server and reactivate the gateways, but no success. The ip addresses have not changed from the gateways and I have entered the code from the app again, but the same problems. In the end I even, as a test, removed both gateways from domoticz, restarted domoticz and added the gateways 1 for 1 again. again the problem: Xiaomi Gateway: receive data from unregistered gateway!
out of misery I went back to Domoticz version 4.9700 when the both gateways worked fine.
Someone an idea what it could be, or the solution? (port 9898 are not closed)

**Comment by kovainfo** [comment]
It's not a bug, it's a feature.
The latest versions of Domoticz have full support for multiple gateways.
but at the same time, the xiaomi devices in the Domoticz should be connected to their gateways accordingly.
this means that if your doorbell is connected to gateway 1, and the lighting is connected to gateway 2, then the devices in the Domoticz should also be connected: the doorbell to gateway 1, and the lighting to gateway 2. And now you can control any xiaomi devices using domotics
Events from xiaomi sensors are transmitted over the network to a single port.
And until all the Domoticz services start, it is possible to receive error messages from an unregistered gateway: it’s just that the second gateway handler has not yet started.

**Comment by kovainfo** [comment]
perhaps you need to change the instructions in the documentation (wiki) for connecting multiple gateways ...

**Comment by vwtune** [comment]
> It's not a bug, it's a feature.
> The latest versions of Domoticz have full support for multiple gateways.
> but at the same time, the xiaomi devices in the Domoticz should be connected to their gateways accordingly.
> this means that if your doorbell is connected to gateway 1, and the lighting is connected to gateway 2, then the devices in the Domoticz should also be connected: the doorbell to gateway 1, and the lighting to gateway 2. And now you can control any xiaomi devices using domotics
> Events from xiaomi sensors are transmitted over the network to a single port.
> And until all the Domoticz services start, it is possible to receive error messages from an unregistered gateway: it’s just that the second gateway handler has not yet started.

But the devices connected to the second gateway, doesn't react or shows there status on domoticz anymore. Only the gateway itself does respons on the doorbell, alarm or rgb settings. but a device, such as a temp sensor or push button, won't react or show there status.

**Comment by kovainfo** [comment]
I’ll ask again: how are the Xiaomi devices configured in the Domoticz?

**Comment by vwtune** [comment]
i don't get your point i guess, but the devices connected to 1 of the gateways, are shown in the devicelist by themself when they are new connected to the gateway, at that point, i can give them a name and add them as a device in domoticz. This is already done for all my devices in the past. So i have devices connected to gateway 1 and gateway 2, but now only the devices connected to gateway 1 can be controlled by domoticz.

**Comment by kovainfo** [comment]
version 4.9700 works just that way. in this version you can only control devices that are connected to 1 gateway.

**Comment by kovainfo** [comment]
In the new version, the sensor settings should correspond to the connection to the gateway. And then you can control any sensors.
<img width="1313" alt="sample" src="https://user-images.githubusercontent.com/36986231/66062185-9b286580-e549-11e9-8f35-14525288a7e2.png">

**Comment by vwtune** [comment]
So what you say, i have to reconfigure my second gateway, and reset all the devices to the second gateway and connect them as a new device to it, then the will show up again in domoticz? Also, i get rid of the error :  Xiaomi Gateway: receive data from unregistered gateway!? ??

**Comment by kovainfo** [comment]
no, not all, but only those that are physically connected to the second gateway.
An error about an unregistered gateway will appear only once at the start of the domotics

**Comment by vwtune** [comment]
okay, i will try to reset the devices connected to the second gateway and then connect them as a new device. Then i hope to see them as a new device in domoticz.
I just have to ignore the error at startup of domoticz then?

**Comment by ricardo777** [comment]
I got the same issue, I tried from removing the xiami gateway, updating the stable to beta and even reinstall domiticz but could not get the issue fixed in the log I have:

2019-12-01 16:12:54.236 Status: XiaomiGateway (ID=9): Delaying worker startup...
2019-12-01 16:12:59.236 Status: XiaomiGateway (ID=9): Selected as main Gateway
2019-12-01 16:12:59.236 Status: Incoming connection from: 10.0.0.40
2019-12-01 16:12:59.236 Status: XiaomiGateway (ID=9): Worker started...
2019-12-01 16:12:59.237 Status: XiaomiGateway (ID=9): Using 10.0.0.75 for local IP address.
2019-12-01 16:13:08.011 Error: XiaomiGateway: received data from unregisted gateway!

How can I solve the issue?

**Comment by kovainfo** [comment]
ricardo777, did you read exactly what was written above? ))

https://github.com/domoticz/domoticz/issues/3649#issuecomment-537437897

**Comment by ricardo777** [comment]
Yes (I hope) I have only 1 gateway, and when I remove the gateway in Domoticz I have no devices left of Xiaomi, when I readd the gateway I get the error.

**Comment by kovainfo** [comment]
> Yes (I hope) I have only 1 gateway, and when I remove the gateway in Domoticz I have no devices left of Xiaomi, when I readd the gateway I get the error.

check the gateway settings in the domiticz hardware panel. IP address, for example.

**Comment by ricardo777** [comment]
Yes ip is correct and can also ping it from Domoticz, also have done a new key generation on the gateway but that did not fixed it also

**Comment by kovainfo** [comment]
if you have only one message at the start of Domotics, it means the gateway handler has not yet loaded. https://github.com/domoticz/domoticz/issues/3649#issuecomment-537437897

**Comment by ricardo777** [comment]
I have this in the beginning of the log:

```
2019-12-01 16:30:18.898 Sunrise: 07:25:00 SunSet: 15:35:00
2019-12-01 16:30:18.898 Day length: 08:10:00 Sun at south: 11:30:00
2019-12-01 16:30:18.898 Civil twilight start: 06:46:00 Civil twilight end: 16:14:00
2019-12-01 16:30:18.898 Nautical twilight start: 06:04:00 Nautical twilight end: 16:57:00
2019-12-01 16:30:18.898 Astronomical twilight start: 05:23:00 Astronomical twilight end: 17:37:00
2019-12-01 16:30:18.943 Active notification Subsystems: (0/13)
2019-12-01 16:30:18.949 Starting shared server on: :::6144
2019-12-01 16:30:18.689 Status: Domoticz V4.10717 (c)2012-2019 GizMoCuz
2019-12-01 16:30:18.689 Status: Build Hash: b38b49e5, Date: 2019-05-09 11:04:08
2019-12-01 16:30:18.700 Status: Startup Path: /home/domoticz/
2019-12-01 16:30:18.939 Status: PluginSystem: Started, Python version '3.6.9'.
2019-12-01 16:30:18.944 Status: WebServer(HTTP) started on address: :: with port 8080
2019-12-01 16:30:18.947 Status: WebServer(SSL) started on address: :: with port 443
2019-12-01 16:30:18.949 Status: TCPServer: shared server started...
2019-12-01 16:30:18.949 Status: RxQueue: queue worker started...
2019-12-01 16:30:20.950 Status: P1MeterTCP: attempt connect to 10.0.0.161:8088
2019-12-01 16:30:20.950 Status: (IKEA Tradfri) Started.
2019-12-01 16:30:20.950 Status: Darksky: Started...
2019-12-01 16:30:20.951 Status: XiaomiGateway (ID=9): Delaying worker startup...
2019-12-01 16:30:20.955 Status: P1MeterTCP: connected to: 10.0.0.161:8088
2019-12-01 16:30:25.951 Status: XiaomiGateway (ID=9): Selected as main Gateway
2019-12-01 16:30:25.951 Status: XiaomiGateway (ID=9): Worker started...
2019-12-01 16:30:25.951 Status: XiaomiGateway (ID=9): Using 10.0.0.160 for local IP address.
2019-12-01 16:30:25.951 Status: EventSystem: reset all events...
2019-12-01 16:30:25.952 Status: EventSystem: reset all device statuses...
2019-12-01 16:30:26.690 (IKEA Tradfri) Connected successfully to: 127.0.0.1
2019-12-01 16:30:26.076 Status: Python EventSystem: Initalizing event module.
2019-12-01 16:30:26.076 Status: EventSystem: Started
2019-12-01 16:30:26.076 Status: EventSystem: Queue thread started...
2019-12-01 16:30:26.401 Status: PluginSystem: Entering work loop.
2019-12-01 16:30:26.614 Status: (IKEA Tradfri) Initialized version 2.0.0, author 'moroen'
2019-12-01 16:30:26.614 Status: (IKEA Tradfri) Entering work loop.
2019-12-01 16:30:31.452 (Weer) Temp + Humidity + Baro (Huidige Weer)
2019-12-01 16:30:31.453 (Weer) Wind (Unknown)
2019-12-01 16:30:31.453 (Weer) UV (UV Index)
2019-12-01 16:30:31.455 (Weer) Rain (Rain)
2019-12-01 16:30:31.456 (Weer) General/Visibility (Unknown)
2019-12-01 16:30:31.456 (Weer) General/Custom Sensor (Ozone Sensor)
2019-12-01 16:30:34.754 Error: XiaomiGateway: received data from unregisted gateway!
2019-12-01 16:30:44.723 Error: XiaomiGateway: received data from unregisted gateway!
2019-12-01 16:30:54.693 Error: XiaomiGateway: received data from unregisted gateway!
```

And 1 gateway in my device list.

I appreciate the help

**Comment by kovainfo** [comment]
It looks like the wrong gateway IP address setting.

**Comment by kovainfo** [comment]
the gateway receives the info packet from the zigbee sensor and sends it to the domoticz along with its IP address. if the IP address does not coincide with any IP address of the gateways registered in the Domoticz, an error message appears.

**Comment by ricardo777** [comment]
Domoticz is running on IP 10.0.0.160 and when checking the Xiaomi gateway via the android app it says ip 10.0.0.223 using that IP in Domoticz and using the standard port 9898

**Comment by kovainfo** [comment]
all information here: https://github.com/domoticz/domoticz/issues/3649#issuecomment-560130191 
Explore your configuration.

**Comment by ricardo777** [comment]
Sorry I really dont get it, I am using the IP of the gateway in the setting in Domoticz I have checked my network but the IP I use is really the IP of the gateway in the device settings.

IP of the gateway and Domoticz vps is set in the DHCP so it is static.

Maybe if the error says the IP of the unregistered gateway it helps

**Comment by ricardo777** [comment]
Update @kovainfo was right in the beginning, I used pktstat -n for checking the ip adresses. I use LXC on Qnap and now I used the IP of the NAS itself even the vm is bridged I needed to use the NAS IP. This finnaly solved the issue. I used the same setup for 2 years but suddenly it did not worked any longer.

**Comment by waltervl** [comment]
Issue can be closed.
